packages/amesos2/example/SimpleSolve_File.cpp

```diff
@@ -262,15 +268,25 @@ int main(int argc, char *argv[]) {
       }
       A->fillComplete();
     }
-    // perform numeric
     //A->describe(*fos,Teuchos::VERB_EXTREME);
-    solver->numericFactorization();
+    // perform numeric
     comm->barrier();
+    {
+      Teuchos::RCP< Teuchos::Time > numerTimer_ = Teuchos::TimeMonitor::getNewCounter ("Time for Numeric Factorization");
+      Teuchos::TimeMonitor numerFactTimer(*numerTimer_);
 
+      solver->numericFactorization();
+      comm->barrier();
```

Comment:
@iyamazaki this calls `comm->barrier();` twice in a row, here and below at (new) L282, is that intended or should one of the barriers be removed?